.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,44 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+    "name": "Python 3",
+    // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+    "image": "mcr.microsoft.com/devcontainers/python:3.12",
+    // Features to add to the dev container. More info: https://containers.dev/features.
+    // "features": {},
+    // Use 'forwardPorts' to make a list of ports inside the container available locally.
+    // "forwardPorts": [],
+    // Use 'postCreateCommand' to run commands after the container is created.
+    "postCreateCommand": "bash ./.devcontainer/scripts/postCreate.sh",
+    "postStartCommand": "bash ./.devcontainer/scripts/postStart.sh",
```

Comment:
Possibly - and this may be a relic from a fix to get devcontainers working for several projects on my machine (Windows with Docker Desktop using WSL2). If someone can confirm that git does not throw permission errors without that line, please feel free to remove it. It is borrowed from another devcontainer that does a bunch of stuff after starting.

Reply:
The postStart.sh is needed in order to install the duckdb pieces. 
I'm working off that shell script so that these happen after the Setup dbt section 

I haven't added fancy logic to detect which OS. 
My assumption is that for Github Codespaces that it is good enough for demoing for the symposium. 
A longer term solution will be needed. 

`
### Setup dbt

echo "Setting up duckdb synthetic data directory"
mkdir ./data

echo "Installing Duckdb in Github Codespaces which is Ubuntu based OS"
wget https://github.com/duckdb/duckdb/releases/download/v1.0.0/duckdb_cli-linux-amd64.zip
unzip duckdb_cli-linux-amd64.zip

echo "Initialize the duckdb file and exit duckdb"
/workspaces/dbt-synthea/duckdb  ./data/synthea_omop_etl.duckdb -s .quit

echo "Debugging dbt to check the connection to duckdb"
dbt debug
echo "Configure dbt dependent Python packages"
dbt deps
echo "Seed dbt duckdb with data"
dbt seed
echo "Compile and Build the dbt project"
dbt compile
dbt build
`

Reply:
Update - My comment above is obviated by Lawrences awesome work to reconcile the python modules so that dbt debug checks both the connector to duckdb and initializes the file specified in the project yaml file.